rotkehlchen/exchanges/kraken.py

```diff
@@ -782,3 +779,282 @@ def query_online_income_loss_expense(
+            )
+            # Group related events
+            raw_events_groupped = defaultdict(list)
+            for raw_event in response:
+                raw_events_groupped[raw_event['refid']].append(raw_event)
+
+            new_events = []
+            for events in raw_events_groupped.values():
+                try:
+                    events = sorted(
+                        events,
+                        key=lambda x: deserialize_timestamp_from_kraken(x['time']),
+                    )
+                except DeserializationError as e:
+                    self.msg_aggregator.add_error(
+                        f'Failed to read timestamp in kraken event group '
+                        f'{events} due to {str(e)}',
+                    )
+                    continue
+                group_events = []
+                for idx, raw_event in enumerate(events):
+                    found_unknown_event = False
+                    try:
+                        timestamp = deserialize_timestamp_from_kraken(raw_event['time'])
+                        identifier = raw_event['refid']
+                        event_type = HistoryEventType.from_string(raw_event['type'])
+                        asset = asset_from_kraken(raw_event['asset'])
+                        event_subtype = None
+                        amount = abs(deserialize_asset_amount(raw_event['amount']))
+                        # If we don't know how to handle an event atm or we find an unsupported
+                        # event type the logic will be to store it as unknown and if in the future
+                        # we need some information from it we can take actions to process them
+                        if event_type == HistoryEventType.TRANSFER:
+                            if raw_event['subtype'] == 'spottostaking':
+                                event_type = HistoryEventType.STAKING
+                                event_subtype = HistoryEventSubType.STAKING_DEPOSIT_ASSET
+                            if raw_event['subtype'] == 'stakingfromspot':
+                                event_type = HistoryEventType.STAKING
+                                event_subtype = HistoryEventSubType.STAKING_RECEIVE_ASSET
+                            if raw_event['subtype'] == 'stakingtospot':
+                                event_type = HistoryEventType.UNSTAKING
+                                event_subtype = HistoryEventSubType.STAKING_REMOVE_ASSET
+                            if raw_event['subtype'] == 'spotfromstaking':
+                                event_type = HistoryEventType.UNSTAKING
+                                event_subtype = HistoryEventSubType.STAKING_RECEIVE_ASSET
+                        elif event_type == HistoryEventType.UNKNOWN:
+                            found_unknown_event = True
+                            self.msg_aggregator.add_warning(
+                                f'Encountered unknown event type {raw_event["type"]}. Event '
```

Comment:
For anything that is user facing message please think quite hard how to phrase it. When a user sees: `"Encountered unknown event type FOO"` in the UI this tells them nothing about where this happened, what might be the cause etc.

Also does not help us when we ask the user what is wrong.

```suggestion
                                f'Encountered unknown kraken historic event type {raw_event["type"]}. Event '
```